core/config.go

```diff
@@ -0,0 +1,72 @@
+package core
+
+import (
+	"net"
+	"net/url"
+	"strings"
+
+	"chain/core/config"
+	"chain/database/sinkdb"
+	"chain/errors"
+)
+
+// Config provides access to Chain Core configuration options
+// and their values.
+func Config(sdb *sinkdb.DB) *config.Options {
+	opts := config.New(sdb)
+
+	equalFirst := func(a, b []string) bool { return a[0] == b[0] }
+	cleanEnclaveTuple := func(tup []string) error {
+		normalized, err := normalizeURL(tup[0])
+		if err != nil {
+			return errors.WithDetailf(err, "Provided URL is invalid: %s", err.Error())
+		}
+		tup[0] = normalized
+		return nil
+	}
+
+	// enclaves defines a set of (URL, access token) tuples to be used
+	// for the local block signer. Tuple equality is defined on the
+	// the URL, not the access token.
+	opts.DefineSet("enclaves", 2, cleanEnclaveTuple, equalFirst)
+
+	return opts
+}
+
+// normalizeURL performs some low-hanging best-effort normalization
+// of the provided URL. See RFC3986, Section 6.
+func normalizeURL(urlstr string) (string, error) {
```

Comment:
instead of this DIY normalizing, we could use [this purell library](https://github.com/PuerkitoBio/purell), but I think we'd need to vendor:

github.com/PuerkitoBio/purell
github.com/PuerkitoBio/urlesc
golang.org/x/net/idna
golang.org/x/text/unicode/norm
golang.org/x/text/width

which feels like a bit much for our needs.